src/EditorFeatures/CSharpTest/NameTupleElement/NameTupleElementTests.cs

```diff
@@ -0,0 +1,234 @@
+        public async Task TestInCall_FirstElement_WithTrivia()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { void M((int arg1, int arg2) x) => M((/*before*/ [||]1 /*after*/, 2)); }",
+@"class C { void M((int arg1, int arg2) x) => M((/*before*/ arg1: 1 /*after*/, 2)); }");
+        }
+
+        [Fact]
+        public async Task TestInCall_FirstElement_Nested()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    int M((int arg1, int arg2) x)
+        => M((M(([||]1, 2)), 3));
+}",
+@"class C
+{
+    int M((int arg1, int arg2) x)
+        => M((M((arg1: 1, 2)), 3));
+}");
+        }
+
+        [Fact]
+        public async Task TestInCall_FirstComma()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { void M((int arg1, int arg2) x) => M((1[||], 2)); }",
+@"class C { void M((int arg1, int arg2) x) => M((arg1: 1, 2)); }");
+        }
+
+        [Fact]
+        public async Task TestInCall_FirstComma2()
+        {
+            await TestMissingAsync(@"class C { void M((int arg1, int arg2) x) => M((1,[||] 2)); }");
+        }
+
+        [Fact]
+        public async Task TestInCall_SecondElement()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { void M((int arg1, int arg2) x) => M((1, [||]2)); }",
+@"class C { void M((int arg1, int arg2) x) => M((1, arg2: 2)); }");
+        }
+
+        [Fact]
+        public async Task TestInCall_CloseParen()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { void M((int arg1, int arg2) x) => M((1, 2[||])); }",
+@"class C { void M((int arg1, int arg2) x) => M((1, arg2: 2)); }");
+        }
+
+        [Fact]
+        public async Task TestUnnamedTuple()
+        {
+            await TestMissingAsync(@"class C { void M((int, int) x) => M(([||]1, 2)); }");
+        }
+
+        [Fact]
+        public async Task TestArrowReturnedTuple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { (int arg1, int arg2, int arg3) M() => ([||]1, 2); }",
+@"class C { (int arg1, int arg2, int arg3) M() => (arg1: 1, 2); }");
+        }
+
+        [Fact]
+        public async Task TestArrowReturnedTuple_LocalFunction()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        (int arg1, int arg2, int arg3) local() => ([||]1, 2);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        (int arg1, int arg2, int arg3) local() => (arg1: 1, 2);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestReturnedTuple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C { (int arg1, int arg2, int arg3) M() { return ([||]1, 2); } }",
+@"class C { (int arg1, int arg2, int arg3) M() { return (arg1: 1, 2); } }");
+        }
+
+        [Fact]
+        public async Task TestReturnedTuple_LongerTuple()
+        {
+            await TestMissingAsync(
+@"class C { (int arg1, int arg2) M() => (1, 2, [||]3); }");
+        }
+    }
+}
```

Comment:
I looked at all tests and could not find one where we add _all_ tuple named arguments, not just one. I personally only add tuple named arguments for all elements or none. Should we consider offering a separate refactoring to add all arguments when the caret is just before or after the open/close parens of the tuple? It can also come as a separate PR...

Reply:
Filed https://github.com/dotnet/roslyn/issues/31274